### PR TITLE
Fix thread enumaration issue on Linux when using Yama

### PR DESCRIPTION
**Pull request by jonwilson030981**
With Yama LSM enabled processes by default can only ptrace their children (unless otherwise configured by sysctl). This appears to be the default setting on Ubuntu Linux. More aggressive changes may be required for other modes.

This patch uses prctl(PR_SET_PTRACER) to give permission to the ptracing thread (in another process) to attach.

- https://www.kernel.org/doc/Documentation/security/Yama.txt
- https://github.com/torvalds/linux/blob/master/security/yama/yama_lsm.c

Note also that Process.enumerateThreads() doesn't raise any exceptions on failure, instead simply returning an empty or reduced set. This may help to identify the cause if subsequent issues are raised. The following issues possibly relate:

- https://github.com/frida/frida/issues/489
- https://github.com/frida/frida-gum/issues/116

### CODE REVIEW

File: gum/backend-linux/gumprocess-linux.c [hunk 1]
```diff
@@ -434,6 +435,10 @@ gum_process_modify_thread (GumThreadId thread_id,
         NULL);
     g_assert (child > 0);
 
+    prctl(PR_SET_PTRACER, child);
```
Comment (oleavr) [inline gum/backend-linux/gumprocess-linux.c]:
```suggestion
    prctl (PR_SET_PTRACER, child);
```

File: gum/backend-linux/gumprocess-linux.c [hunk 2]
```diff
@@ -434,6 +435,10 @@ gum_process_modify_thread (GumThreadId thread_id,
...
+
+    gum_put_ack(fd, GUM_ACK_PTRACER);
```
Comment (oleavr) [inline gum/backend-linux/gumprocess-linux.c]:
```suggestion
    gum_put_ack (fd, GUM_ACK_PTRACER);
```

File: gum/backend-linux/gumprocess-linux.c [hunk 3]
```diff
@@ -434,6 +435,10 @@ gum_process_modify_thread (GumThreadId thread_id,
         NULL);
     g_assert (child > 0);
 
+    prctl(PR_SET_PTRACER, child);
+
```
Comment (oleavr) [inline gum/backend-linux/gumprocess-linux.c]:
```suggestion
```

**Comment by jonwilson030981** [comment]
Sorry, old coding style habits die hard! Hope this is better and by forcing an amended commit I don't make the merge hard!

**Comment by oleavr** [comment]
No worries! Thanks for digging into and fixing this 🤘

**Pull request merged**